added README.md and support for specifying subscription channel (#11)
rh-ecosystem-edge/nvidia-ci: README.md

Ecosystem Edge NVIDIA-CI - Golang Automation CI
=======
# NVIDIA-CI

## Overview

> The [nvidia-ci](https://github.com/rh-ecosystem-edge/nvidia-ci) is an automation/CI framework to test NVIDIA operators, starting with the GPU Operator.

Reviewer: ```suggestion
This repository is an automation/CI framework to test NVIDIA operators, starting with the GPU Operator.
```

Author: Updated.

This project is based on golang + [ginkgo](https://onsi.github.io/ginkgo) framework.

### Project requirements
* golang v1.20.x

> * ginkgo v2.x

Reviewer: How about we refer people to the `go.mod` file for getting the requirements?
This part of the README is going to be outdated very fast and probably forgotten.

Author: Good idea.  Updated

> ## nvidia-ci

Reviewer: Should we move that entire section [here](https://github.com/rh-ecosystem-edge/nvidia-ci/pull/11/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R5)?

Author: Updated

> The [nvidia-ci](https://github.com/rh-ecosystem-edge/nvidia-ci) project is designed to test a pre-installed OpenShift Container Platform (OCP) cluster which meets the following requirements:

Reviewer: ```suggestion
The framework in this repository is designed to test NVIDIA's operators on a pre-installed OpenShift Container Platform (OCP) cluster which meets the following requirements:
```

### Mandatory setup requirements:
* OCP cluster installed with version >=4.12

### Supported setups:

> * Regular cluster 3 master nodes (VMs or BMs) 2 workers (VMs or BMs)

Reviewer: Aren't we supporting 3 workers instead of 2?

Author: This should read minimum 2 worker nodes and 3 master nodes.  So 3 worker nodes should be supported.

Author: Updated

Reviewer: Seems like this one isn't resolved yet.

* Single Node Cluster (VM or BM)
* Public Clouds (AWS, GCP and Azure)

### General environment variables
#### Mandatory:
* `KUBECONFIG` - Path to kubeconfig file.
#### Optional:
* Logging with glog

We use glog library for logging. In order to enable verbose logging the following needs to be done:

1. Make sure to import inittool package in your go script, per this example:

<sup>
    import (
      . "github.com/rh-ecosystem-edge/nvidia-ci/tests/internal/inittools"
    )
</sup>

2. Need to export the following SHELL variable:
> export VERBOSE_LEVEL=100

##### Notes:

  1. The value for the variable has to be >= 100.
  2. The variable can simply be exported in the shell where you run your automation.

>   3. The go file you work on has to be in a directory under github.com/rh-ecosystem-edge/nvidia-ci/tests/ directory for being able to import inittools.

Reviewer: Are you sure about it? The instructions [here](https://github.com/rh-ecosystem-edge/nvidia-ci/pull/11/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R34-R38) are pulling the library directly from github (and not locally).

Author: We are importing the inittools package from Github

  4. Importing inittool also initializes the apiclient and it's available via "APIClient" variable.

* Collect logs from cluster with reporter

We use k8reporter library for collecting resource from cluster in case of test failure.
In order to enable k8reporter the following needs to be done:

1. Export DUMP_FAILED_TESTS and set it to true. Use example below
> export DUMP_FAILED_TESTS=true

2. Specify absolute path for logs directory like it appears below. By default /tmp/reports directory is used.
> export REPORTS_DUMP_DIR=/tmp/logs_directory

> <!-- TODO Update this section with optional env vars for each test suite -->

Reviewer: Can we add a Jira task for it?

Author: Removed the TODO line since the paramaters described below show which ones are optional and which ones are required.

## How to run

The test-runner [script](scripts/test-runner.sh) is the recommended way for executing tests.

Parameters for the script are controlled by the following environment variables:

> - `TEST_FEATURES`: list of features to be tested.   subdirectories under tests that match a feature will be included (internal directories are excluded) - _required_

Reviewer: Are we using a `,` separator between the different tests in the variable name? We should probably mention it in the doc.

Author: Added it.
